### ISSUE TITLE
Uncaught Exception in NodeJS support

### ISSUE THREAD

**Issue by jvanburen** [cs-au-dk/TAJS#11]
Analyzing the following file with `java -jar dist/tajs-all.jar -nodejs test-node.js` raises an uncaught exception (reproduced below):
`test-node.js`:
```javascript
require('os');
```
The exception (compiled by changing `debug` to `on` in `build.xml`):
```
Exception in thread "main" java.nio.file.ProviderMismatchException
	at sun.nio.fs.UnixPath.toUnixPath(UnixPath.java:200)
	at sun.nio.fs.UnixPath.relativize(UnixPath.java:410)
	at sun.nio.fs.UnixPath.relativize(UnixPath.java:43)
	at dk.brics.tajs.util.PathAndURLUtils.getRelativeTo(PathAndURLUtils.java:123)
	at dk.brics.tajs.util.PathAndURLUtils.getRelativeToWorkingDirectory(PathAndURLUtils.java:116)
	at dk.brics.tajs.flowgraph.SourceLocation.toUserFriendlyString(SourceLocation.java:255)
	at dk.brics.tajs.flowgraph.SourceLocation.toString(SourceLocation.java:174)
	at dk.brics.tajs.js2flowgraph.FlowGraphBuilder.reportParseMessages(FlowGraphBuilder.java:179)
	at dk.brics.tajs.js2flowgraph.FlowGraphBuilder.makeAST(FlowGraphBuilder.java:165)
	at dk.brics.tajs.js2flowgraph.FlowGraphBuilder.transformFunctionBody(FlowGraphBuilder.java:616)
	at dk.brics.tajs.js2flowgraph.FlowGraphMutator.addTopLevelFunction(FlowGraphMutator.java:181)
	at dk.brics.tajs.js2flowgraph.FlowGraphMutator.extendFlowGraphWithTopLevelFunction(FlowGraphMutator.java:151)
	at dk.brics.tajs.analysis.nativeobjects.FunctionFileLoader.loadFunction(FunctionFileLoader.java:89)
	at dk.brics.tajs.analysis.nativeobjects.TAJSFunctionEvaluator.lambda$makeImplementations$29(TAJSFunctionEvaluator.java:521)
	at dk.brics.tajs.analysis.nativeobjects.TAJSFunctionEvaluator$TAJSFunctionImplementation.transfer(TAJSFunctionEvaluator.java:795)
	at dk.brics.tajs.analysis.nativeobjects.TAJSFunctionEvaluator.evaluate(TAJSFunctionEvaluator.java:129)
	at dk.brics.tajs.analysis.HostAPIs.evaluate(HostAPIs.java:63)
	at dk.brics.tajs.analysis.FunctionCalls.lambda$callFunction$0(FunctionCalls.java:347)
	at dk.brics.tajs.solver.GenericSolver$SolverInterface.lambda$withState$0(GenericSolver.java:119)
	at dk.brics.tajs.solver.GenericSolver$SolverInterface.withState(GenericSolver.java:109)
	at dk.brics.tajs.solver.GenericSolver$SolverInterface.withState(GenericSolver.java:118)
	at dk.brics.tajs.analysis.FunctionCalls.callFunction(FunctionCalls.java:338)
	at dk.brics.tajs.analysis.js.NodeTransfer.visit(NodeTransfer.java:627)
	at dk.brics.tajs.flowgraph.jsnodes.CallNode.visitBy(CallNode.java:255)
	at dk.brics.tajs.analysis.Transfer.visit(Transfer.java:73)
	at dk.brics.tajs.flowgraph.jsnodes.Node.visitBy(Node.java:42)
	at dk.brics.tajs.analysis.Transfer.transfer(Transfer.java:62)
	at dk.brics.tajs.solver.GenericSolver.solve(GenericSolver.java:372)
	at dk.brics.tajs.Main.run(Main.java:296)
	at dk.brics.tajs.Main.main(Main.java:93)
```
It's caused because there are some node.js modules modeled, but they are compiled into the jar, so they're not in the same filesystem as the file we analyze, and getting the relative path makes no sense.

**Comment by amoeller** [comment]
Try changing
https://github.com/cs-au-dk/TAJS/blob/master/src/dk/brics/tajs/flowgraph/SourceLocation.java#L249
from
```
if (location.getProtocol().equalsIgnoreCase("http") || location.getProtocol().equalsIgnoreCase("https")) { 
```
to
```
if (location.getProtocol().equalsIgnoreCase("http") || location.getProtocol().equalsIgnoreCase("https") || location.getProtocol().equalsIgnoreCase("jar")) {
```

**Comment by amoeller** [comment]
Should be fixed in 3244c9be
